tests/v2/validation/certrotation/cert_rotation_test.go

```diff
@@ -38,10 +42,29 @@ func (r *V2ProvCertRotationTestSuite) SetupSuite() {
 }
 
 func (r *V2ProvCertRotationTestSuite) TestCertRotation() {
-	r.Run("test-cert-rotation", func() {
-		require.NoError(r.T(), rotateCerts(r.client, r.client.RancherConfig.ClusterName))
-		require.NoError(r.T(), rotateCerts(r.client, r.client.RancherConfig.ClusterName))
-	})
+	id, err := clusters.GetV1ProvisioningClusterByName(r.client, r.client.RancherConfig.ClusterName)
+	require.NoError(r.T(), err)
+
+	cluster, err := r.client.Steve.SteveType(ProvisioningSteveResouceType).ByID(id)
+	require.NoError(r.T(), err)
+
+	spec := &provv1.ClusterSpec{}
+	err = steveV1.ConvertToK8sType(cluster.Spec, spec)
+	require.NoError(r.T(), err)
+
+	if strings.Contains(spec.KubernetesVersion, "-rancher") || len(spec.KubernetesVersion) == 0 {
+		r.Run("test-cert-rotation RKE1", func() {
+			require.NoError(r.T(), rotateRKE1Certs(r.client, r.client.RancherConfig.ClusterName))
+			require.NoError(r.T(), rotateRKE1Certs(r.client, r.client.RancherConfig.ClusterName))
+		})
+
+	} else {
+		r.Run("test-cert-rotation "+strings.Split(spec.KubernetesVersion, "+")[1], func() {
```

Comment:
I don't feel overly strongly about this, but how do you feel of having the name RKE2 and K3s?

Reply:
updated to only contain rke2 or k3s